glitter_news/views.py

```diff
@@ -44,6 +44,18 @@ class PostDetailView(GlitterDetailMixin, DateDetailView):
     month_format = '%m'
     date_field = 'date'
 
+    def get_allow_future(self):
+        """
+        Only superusers and users with the permission can edit the post.
+        """
+        qs = self.get_queryset()
+        post_edit_permission = '{}.edit_{}'.format(
+            qs.model._meta.app_label, qs.model._meta.model_name
+        )
+        if self.request.user.is_superuser or self.request.user.has_perm(post_edit_permission):
```

Comment:
Do we need the additional check for a superuser? As has_perm will always return True for superusers.

Up to you!